core/src/test/java/org/hibernate/ogm/backendtck/callbacks/PreUpdateTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.callbacks;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PreUpdateTest extends OgmJpaTestCase {
+
+	private static final String INITIAL = "initial";
+	private static final String UPDATED = "updated";
+
+	private EntityManager em;
+
+	@Before
+	public void setUp() {
+		em = getFactory().createEntityManager();
+	}
+
+	@After
+	@Override
+	public void removeEntities() throws Exception {
+		em.getTransaction().begin();
+		em.remove( em.find( PreUpdatableBus.class, 1 ) );
+		em.getTransaction().commit();
+		em.close();
+	}
+
+	/**
+	 * Update an entity which uses a @PreUpdate annotated method
+	 * to set boolean field to 'true'.
+	 */
+	@Test
+	public void testFieldSetInPreUpdate() {
+		em.getTransaction().begin();
+
+		PreUpdatableBus bus = new PreUpdatableBus();
+		bus.setId( 1 );
+		bus.setField( INITIAL );
+
+		em.persist( bus );
+		em.getTransaction().commit();
+		em.clear();
+
+		em.getTransaction().begin();
+
+		bus = em.find( PreUpdatableBus.class, bus.getId() );
+
+		assertNotNull( bus );
+		assertEquals( bus.getField(), INITIAL );
+		assertFalse( bus.isPreUpdated() );
+
+		bus.setField( UPDATED );
+		em.merge( bus );
```

Comment:
FYI: There is no need to merge in this scenario. The entity is managed and therefore it will be updated

Reply:
Yes. You right!  it is item N3.2 of specification. I will correct it.